Plutarch/Builtin.hs

```diff
@@ -154,11 +163,13 @@ data PAsData (a :: PType) (s :: S)
 
 data PAsDataLifted (a :: PType)
 
-instance PUnsafeLiftDecl (PAsDataLifted a) (PAsData a) where
-  type PLifted (PAsData a) = PAsDataLifted a
-  type PLiftedRepr (PAsData a) = Data
-  pliftToRepr = \case {}
-  pliftFromRepr _ = Nothing
+instance PConstant (PAsDataLifted a) where
+  type PConstantRepr (PAsDataLifted a) = Data
+  type PConstanted (PAsDataLifted a) = PAsData a
+  pconstantToRepr = \case {}
+  pconstantFromRepr _ = Nothing
+
+instance PUnsafeLiftDecl (PAsData a) where type PLifted (PAsData a) = PAsDataLifted a
```

Comment:
doesn't this mean we can't actually use `pconstant`? Or is that deliberate...

Reply:
No, you can, it's just not obvious. You can do e.g. `pconstant []` for a list of this, but you can't make a non-empty list. This is a slight but important difference, because it means you can make `pnil` work.

Reply:
The instance should probably be overlappable.